### DISCUSSION TITLE
Bunks - two sleeping places for two crew member.

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#7448] [Ideas and suggestions]
It’s the right thing to do.
![Image 008 (1)](https://user-images.githubusercontent.com/93348499/142929920-392d4604-beff-45ab-aa67-ae89c26fcd83.png)
Putting two beds together in an editor is not the best solution.

**Comment by MaaiG** [comment]
\Content\Map\Outposts\OutpostItems.xml

Bunks (opdeco_bunks1): 160
upper bunk IMHO
`<limbposition limb="Head" position="14,-140" />`
`<limbposition limb="Torso" position="124,-110" />`
`<limbposition limb="Waist" position="264,-130" />`
`<limbposition limb="RightFoot" position="484,-130" />`
`<limbposition limb="LeftFoot" position="484,-130" />`
`<limbposition limb="RightHand" position="264,-130" allowusinglimb="false" />`
`<limbposition limb="LeftHand" position="264,-130" allowusinglimb="false" />`

Military Bunk (opdeco_bunkbeds): 135
upper bunk
`<limbposition limb="Head" position="-26,-35" />`
`<limbposition limb="Torso" position="84,-5" />`
`<limbposition limb="Waist" position="224,-25" />`
`<limbposition limb="RightFoot" position="444,-25" />`
`<limbposition limb="LeftFoot" position="444,-25" />`
`<limbposition limb="RightHand" position="224,-25" allowusinglimb="false" />`
`<limbposition limb="LeftHand" position="224,-25" allowusinglimb="false" />`

**Comment by MaaiG** [comment]
Captain's log, star date 2021D20
...
_Looks like we’ve stumbled on a strange anomaly. There may be some sort of scum attached to the boat... we haven’t figured it out yet. At some point it was discovered that only one crew member could sleep at the same time! The other one just could not lie down on a bed. It’s obviously some kind of mental thing... Maybe there’s a shard of something alien in the hull.
So far, no one’s been hurt, but I’ve had to change the duty schedule and the lack of sleep is starting to come back. Insomnia is a dangerous thing._
...

**Comment by DrBruhman** [comment]
https://github.com/FakeFishGames/Barotrauma/discussions/15464
tried making a mod that makes this but encountered a oversight from the devs

**Reply by DrBruhman** [reply]
mod:https://steamcommunity.com/sharedfiles/filedetails/?id=3398737146

**Comment by pavel4win** [comment]
I think the bed sprites need to be replaced so as not to mislead players.
If you need a double bed, place two medical beds one above the other to save space.
Also, the medical bed is 50% more efficient.

**Comment by Ivanzyfer** [comment]
Why hasn't it been added to the game yet?

**Reply by SomeRandomNoobKekeke** [reply]
Because it was modded into the game 100 years ago https://github.com/FakeFishGames/Barotrauma/discussions/7448#discussioncomment-11717631

**Reply by Ivanzyfer** [reply]
fuuu, modifications

**Reply by Ivanzyfer** [reply]
The main functions of the game should not be modified. The modifications should have Warhammer 40,000 skins and pillows waifu.

**Comment by NoahRoopull** [comment]
Adding a second Controller component (activated with a different key) does the trick if the original items are overridden (as seen in Bunks Fix), but this still causes some issues with bots.

They assume that there's only one controller per item (given that it's a safe assumption in vanilla), so if another character begins using any controller (even a different one), they run CancelUsing, which stops them from using the Controller and abandons orders to use said item (e.g. a command to operate/sleep in a bunk or other controller). Some changes have to be made either to Controller's Select method or only players will be able to use both bunks at once.

Additionally, perhaps beds should be given the "chair" tag or something similar so bots will automatically use them while idle?

Edit: Clarification

**Reply by DrBruhman** [reply]
no shit, sherlock

**Reply by Ivanzyfer** [reply]
If I understood the problem correctly, then you can simply divide the bed into 2 parts, each will be independent of each other and they can be used without changes in the code.
